src/Algebra/Morphism/Structures.agda

```diff
@@ -0,0 +1,111 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Morphisms between algebraic structures
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary.Core
+
+module Algebra.Morphism.Structures
+  {a ℓ₁} {A : Set a} (_≈₁_ : Rel A ℓ₁) -- Equality over the domain
+  {b ℓ₂} {B : Set b} (_≈₂_ : Rel B ℓ₂) -- Equality over the codomain
+  where
+
+open import Algebra.Core
+open import Algebra.Morphism.Definitions A B _≈₂_
+open import Level using (_⊔_)
+open import Function.Definitions _≈₁_ _≈₂_
+open import Relation.Binary.Morphism.Structures
+
+------------------------------------------------------------------------
+-- Morphisms over magma-like structures
+------------------------------------------------------------------------
+
+module _ (_∙_ : Op₂ A) (_◦_ : Op₂ B) (⟦_⟧ : A → B) where
+
+  record IsMagmaHomomorphism : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
+    field
+      isRelHomomorphism : IsRelHomomorphism _≈₁_ _≈₂_ ⟦_⟧
+      homo              : Homomorphic₂ ⟦_⟧ _∙_ _◦_
+
```

Comment:
I would expect a MagmaHomomorphism to be between two magmas.
Why are the bundles unpacked here?

Reply:
So yes, we could parameterise the structures with `RawMagma`, `RawMonoid`, `RawRing` etc. This would be much neater, particularly once we get up to rings, but I wasn't sure whether consistency was better as we don't have the corresponding "raw" structures for the `Relation.Binary.Morphism` hierarchy.

Reply:
If this is possible, I also would be in favor of parametrizing these algebraic morphisms by the raw structures, even if this is inconsistent with morphisms of relations.

Note that they cannot be parametrized by the non-raw structures `Magma` or `Monoid` in their current form, as the main purpose of `Algebra.Morphism.MagmaMonomorphism` is to prove that the domain of a monomorphism is a magma, etc. – although one could argue this is part of the definition of morphisms of magmas or monoids. On the other hand, one might argue that this rather is part of the definition of the corresponding category and not necessary in the definition of the morphisms themselves.